packages/contracts-periphery/test/contracts/L1/messaging/L1ERC721Bridge.spec.ts

```diff
@@ -0,0 +1,370 @@
+
+const ERR_INVALID_MESSENGER = 'OVM_XCHAIN: messenger contract unauthenticated'
+const ERR_INVALID_X_DOMAIN_MSG_SENDER =
+  'OVM_XCHAIN: wrong sender of cross-domain message'
+const ERR_ALREADY_INITIALIZED = 'Contract has already been initialized.'
+const DUMMY_L2_ERC721_ADDRESS = ethers.utils.getAddress(
+  '0x' + 'abba'.repeat(10)
+)
+const DUMMY_L2_BRIDGE_ADDRESS = ethers.utils.getAddress(
+  '0x' + 'acdc'.repeat(10)
+)
+
+const FINALIZATION_GAS = 1_200_000
+
+describe('L1ERC721Bridge', () => {
+  // init signers
+  let l1MessengerImpersonator: Signer
+  let alice: Signer
+  let bob: Signer
+  let bobsAddress
+  let aliceAddress
+  let tokenId
+  let aliceInitialBalance
+
+  // we can just make up this string since it's on the "other" Layer
+  let Factory__L1ERC721: MockContractFactory<ContractFactory>
+  let IL2ERC721Bridge: Interface
+  before(async () => {
+    ;[l1MessengerImpersonator, alice, bob] = await ethers.getSigners()
+
+    // deploy an ERC721 contract on L1
+    Factory__L1ERC721 = await smock.mock(
+      '@openzeppelin/contracts/token/ERC721/ERC721.sol:ERC721'
+    )
+
+    // get an L2ERC721Bridge Interface
+    IL2ERC721Bridge = (await ethers.getContractFactory('L2ERC721Bridge'))
+      .interface
+
+    aliceAddress = await alice.getAddress()
+    bobsAddress = await bob.getAddress()
+    aliceInitialBalance = 5
+    tokenId = 10
+  })
+
+  let L1ERC721: MockContract<Contract>
+  let L1ERC721Bridge: Contract
+  let Fake__L1CrossDomainMessenger: FakeContract
+  beforeEach(async () => {
+    // Get a new mock L1 messenger
+    Fake__L1CrossDomainMessenger = await smock.fake<Contract>(
+      new ethers.utils.Interface(ICrossDomainMessenger.abi),
+      { address: await l1MessengerImpersonator.getAddress() } // This allows us to use an ethers override {from: Fake__L1CrossDomainMessenger.address} to mock calls
+    )
+
+    // Deploy the contract under test
+    L1ERC721Bridge = await (
+      await ethers.getContractFactory('L1ERC721Bridge')
+    ).deploy()
+    await L1ERC721Bridge.initialize(
+      Fake__L1CrossDomainMessenger.address,
+      DUMMY_L2_BRIDGE_ADDRESS
+    )
+
+    L1ERC721 = await Factory__L1ERC721.deploy('L1ERC721', 'ERC')
+
+    await L1ERC721.setVariable('_owners', {
+      [tokenId]: aliceAddress,
+    })
+    await L1ERC721.setVariable('_balances', {
+      [aliceAddress]: aliceInitialBalance,
+    })
+  })
+
+  describe('initialize', () => {
+    it('Should only be callable once', async () => {
+      await expect(
+        L1ERC721Bridge.initialize(
+          ethers.constants.AddressZero,
+          DUMMY_L2_BRIDGE_ADDRESS
+        )
+      ).to.be.revertedWith(ERR_ALREADY_INITIALIZED)
+    })
+  })
+
+  describe('ERC721 deposits', () => {
+    beforeEach(async () => {
+      await L1ERC721.connect(alice).approve(L1ERC721Bridge.address, tokenId)
+    })
+
+    it('depositERC721() escrows the deposit and sends the correct deposit message', async () => {
```

Comment:
Can you add an assertion on the event being emitted for the `depositERC721` tests?